src/client/executor.rs

```diff
@@ -72,6 +72,15 @@ impl Client {
         }
         match session.into() {
             Some(session) => {
+                if !Arc::ptr_eq(&self.inner, &session.client().inner) {
```

Comment:
We don't have `PartialEq` derived for `Client` (and it wouldn't make much sense to do so), so this verifies that the `inner` fields on both the clients point to the same data.